src/Aliases.jl

```diff
@@ -83,3 +83,4 @@ Base.@deprecate_binding DirectSum direct_sum
 # Deprecated in 0.34.*
 Base.@deprecate_binding PuiseuxSeriesField puiseux_series_field
 Base.@deprecate_binding PuiseuxSeriesRing puiseux_series_ring
+Base.@deprecate_binding istrivial is_trivial
```

Comment:
Couldn't we remove almost *all* `@deprecate_binding` with that argument, as for most of them we had multiple breaking releases since we deprecated?

But  that is only if one believes that semver magically solves the problem of compatibility. But it doesn't: users still have code using the old names; if their code uses `istrivial`, they'll notice if it is gone, semver or not. On the other hand, I am convinced virtually nobody will care or even notice about GroupsCore not being involved anymore...